_test/test_utilities/test_range_add_border.m

```diff
@@ -0,0 +1,91 @@
+classdef test_range_add_border < TestCase
+    % Unit tests to check range_add_border function
+    
+    properties
+    end
+    
+    methods
+        function ps = test_range_add_border(varargin)
+            if nargin>0
+                name = varargin{1};
+            else
+                name = 'test_range_add_border ';
+            end
+            ps = ps@TestCase(name);
+        end
+        %
+        function test_zero_border_do_nothing(~)
+            range = [-1*ones(1,4);ones(1,4)];
+            range_out = range_add_border(range,0);
+            assertEqual(range,range_out);
+        end
+        %
+        function test_no_tol_eq_eps_tol(~)
+            range = [-1*ones(1,4);ones(1,4)];
+            range_no  = range_add_border(range);
+            range_eps = range_add_border(range,-eps);
+            assertEqual(range_no,range_eps );
+        end
+        %
+        function test_zero_width(~)
+            range = [-1,1,-1,1;-1,1,-1,1];
+            range_zero  = range_add_border(range);
+            fe = eps;
+            ref_range = [-1-fe,1-fe,-1-fe,1-fe;
+                -1+fe,1+fe,-1+fe,1+fe];
+            assertEqual(range_zero,ref_range);
+        end
+        %
+        function test_near_zero(~)
+            range = [0,0,0,0;0,1,0,1];
+            range_zero  = range_add_border(range);
+            fe = eps;
+            ref_range = [-fe,-fe,-fe,-fe;
+                fe,1+fe,fe,1+fe];
+            assertEqual(range_zero,ref_range);
+        end
+        %
+        function test_large_value_relerr(~)
+            range = [-100,-100;100,100];
+            range_calc  = range_add_border(range);
+            fe = eps;
+            ref_range = [-100-100*fe,-100-100*fe;...
+                100+100*fe,100+100*fe];
+            assertEqual(range_calc,ref_range);
+        end
+        %
+        function test_large_value_abserr_ignored(~)
```

Comment:
why is an eps sized abs border ignored?

Reply:
what is large? when does the behaviour from adding eps to not adding eps happen?

Reply:
it is known issue with eps and large numbers as eps is minimal value so 1+eps ~=1; 10+eps == 10. This is expected behavior -- tested alongside with large absolute value, zero size range, as for  zero size range it still has to be eps between the ranges.

Reply:
it just tests expected behavior. Large values here will probably start from 2, but may be processor-dependent